### PR TITLE
ENH: Added parameter output_axis_rule to CRS.to_wkt

### PR DESCRIPTION
**Pull request by jjimenezshaw**
This PR adds the parameter `output_axis_rule` to the method `CRS.to_wkt`, as documented in https://proj.org/development/reference/functions.html#c.proj_as_wkt

`YES` is mapped to `True`,
 `NO` to `False`,
 and `AUTO` to `None` (default).

 - [x] Tests added
 - [x] Fully documented, including `history.rst` for all changes and `api/*.rst` for new API

### CODE REVIEW

File: pyproj/_crs.pyx [hunk 1]
```diff
@@ -417,12 +425,17 @@ cdef class Base:
             The version of the WKT output.
         pretty: bool, default=False
             If True, it will set the output to be a multiline string.
+        output_axis_rule: bool, optional, default=None
+            If True, it will set the axis rule on any case. If false, never.
+            None for AUTO, that depends on the CRS and version.
+            .. versionadded:: 3.5.1
```
Comment (snowman2) [inline pyproj/_crs.pyx:431]: Instead, I recommend adding this above parameters:

```
        .. versionadded:: 3.6.0 output_axis_rule
```

```suggestion
            .. versionadded:: 3.5.1
```

Comment (jjimenezshaw) [reply pyproj/_crs.pyx]: done

Comment (snowman2) [reply pyproj/_crs.pyx]: Looks great. Mind also doing that in crs.py?

Comment (jjimenezshaw) [reply pyproj/_crs.pyx]: :man_facepalming: yes, of course.

**Comment by snowman2** [comment]
Thanks @jjimenezshaw :+1:

**Pull request merged**